torch/_functorch/aot_autograd.py

```diff
@@ -1219,25 +1225,72 @@ def call_func_with_args(f, args, steal_args=False, disable_amp=False):
             del guard
     return out
 
-def aot_dispatch_base(flat_fn, flat_args: List[Tensor], aot_config: AOTConfig, *, fw_metadata: ViewAndMutationMeta):
+@dataclasses.dataclass
+class SubclassCreationMeta:
+    """
+    Used for AOTDispatch: The graph that we'd like to trace out contains flat tensor inputs,
+    And this dataclass gives us the information we need to reconstruct a tensor subclass
+    from our flat inputs.
+    """
+
+    creation_lambda: Callable[..., torch.Tensor]
+    flat_tensor_start_idx: int
```

Comment:
I'm not a fan of all of the index swizzling. In particular, why can't we keep a structured, nested data structure of input tensors, and only flatten at the *very end*?